pkg/backup/backup.go

```diff
@@ -308,12 +309,13 @@ func (kb *kubernetesBackupper) BackupWithResolvers(
 		}
 	}
 
-	ctx, cancelFunc := context.WithTimeout(context.Background(), podVolumeTimeout)
-	defer cancelFunc()
+	var podVolumeCancelFunc context.CancelFunc
+	kb.podVolumeContext, podVolumeCancelFunc = context.WithTimeout(context.Background(), podVolumeTimeout)
```

Comment:
I have two concerns about this change:
* First, Golang suggests not to store the context in a structure. This change is opposite to the suggestion. https://pkg.go.dev/context#pkg-overview
* Second, `kubernetesBackupper` is global to the Velero server instance. Initializing the structure's parameter in the function `BackupWithResolvers `, which is called per backup, seems not a good practice.

Reply:
We keep the context into the backupper structure for two reasons:
1. the PVB/PVR context is not the primary context of the backupper, it is only used by the PVB/PVR part of the backupper
2. The timeout/context is used by multiple functions in different levels, so we have to initialize the context at the beginning of the backup/restore so as to make sure the 4 hour timeout works correctly

And for the two concerns:
For 1, the suggestion in https://pkg.go.dev/context#pkg-overview is a high level suggestion to make sure a clear lifecycle of context. Here, we already assure the lifecycle as same as the single backup.
For 2, the context could be regarded as part of the runtime context of the backupper instance, so its lifecycle isn't the same as the backupper necessarily.